library/ssl_tls13_client.c

```diff
@@ -1153,6 +1165,32 @@ int mbedtls_ssl_tls13_write_client_hello_exts( mbedtls_ssl_context *ssl,
     }
 #endif
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+    if( mbedtls_ssl_conf_tls13_some_psk_enabled( ssl ) &&
+        ( mbedtls_ssl_conf_has_static_psk( ssl->conf ) == 1
+#if defined(MBEDTLS_SSL_SESSION_TICKETS)
```

Comment:
Seems early data can work with external psk and session tickets psk, why should we bind the MBEDTLS_SSL_SESSION_TICKETS with MBEDTLS_SSL_EARLY_DATA together?

Reply:
We reach agreement to support PSK, but Ronald did not agree with that. 
And see https://github.com/Mbed-TLS/mbedtls/pull/6486#discussion_r1013533945. PSK is optional, but session ticket is MUST to prevent replay risk.

Reply:
How about your opinion?  Should we add a check about `#if defined(MBEDTLS_SSL_EARLY_DATA) && !defined(MBEDTLS_SSL_SESSION_TICKETS) in check_config.h` as jerry said above? @ronald-cron-arm